arduino/libraries/WiFi/src/WiFi.cpp

```diff
@@ -697,6 +703,15 @@ void WiFiClass::handleSystemEvent(system_event_t* event)
     case SYSTEM_EVENT_AP_START: {
       struct netif* apNetif;
 
+      mdns_init();
```

Comment:
Are you sure this `mdns_init` is needed? I believe without it macOS was fine in STA mode, should we also port to STA mode?

Reply:
As described by the example seems yes, its recommend by the example, i had test adding in the properly interface like how we do in STA but not works i had found an issue on the epressif repo.
This is why i suppose to test the mdns and from the result was that works for the access point.
We can try to move on other direction implement in STA mode and removing the interface management if a plus.

Reply:
here the example: https://github.com/espressif/esp-idf/blob/master/examples/protocols/mdns/main/mdns_example_main.c

Reply:
As i understand when the AP interface is used we should enable a dns local instance because we don't have a reference DNS Server like the STA (i'm testing it).
Now i'm try to correctly configure the local dns and ad the feature `AS_DYNAMIC` in order to be able to manage the lookup table and the new rules if this works we could leave the idea  of implement the mdns.